modules/system/assets/js/snowboard/abstracts/PluginBase.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Plugin base abstract.
+ *
+ * This class provides the base functionality for all plugins.
+ *
+ * @copyright 2021 Winter.
+ * @author Ben Thomson <git@alfreido.com>
+ */
+export default class PluginBase {
+    /**
+     * Constructor.
+     *
+     * The constructor is provided the Snowboard framework instance.
+     *
+     * @param {Snowboard} snowboard
+     */
+    constructor(snowboard) {
+        this.snowboard = snowboard;
```

Comment:
Should this reference be `this.Snowboard` to match the public API or am I overthinking it?

Reply:
Not technically - `this.snowboard` is a reference to the global Snowboard instance, but is injected into all plugin instances like Laravel does with the Application class. The standard is for properties to be camelCase, so `snowboard` is correct here.